multirom/multirom.cpp

```diff
@@ -177,8 +182,8 @@ bool MultiROM::setRomsPath(std::string loc)
 		return false;
 	}
 
-	m_curr_roms_path = "/mnt/multirom-"TARGET_DEVICE"/";
-	mkdir("/mnt/multirom-"TARGET_DEVICE"/", 0777);
+    m_curr_roms_path = "/mnt/multirom-"  + (string)TARGET_DEVICE + "/";
```

Comment:
Don't cast it to string, just put a space in between the quotatuion marks and TARGET_DEVICE define.